state/upgrades.go

```diff
@@ -1884,3 +1885,54 @@ func UpdateInheritedControllerConfig(pool *StatePool) error {
 	}
 	return nil
 }
+
+// EnsureDefaultModificationStatus ensures that there is a modification status
+// document for every machine in the statuses.
+func EnsureDefaultModificationStatus(pool *StatePool) error {
+	st := pool.SystemState()
+	db := st.db()
+
+	machineCol, machineCloser := db.GetRawCollection(machinesC)
+	defer machineCloser()
+	// Find all machines that are alive.
+	machineIter := machineCol.Find(bson.D{{"life", life.Alive}}).Iter()
```

Comment:
I'm not sure what the assertions are like for the removal of the status docs for the machine when it gets removed, so I'm not clear whether we should be adding it for all machines or just the alive ones.

Reply:
The removeStatusOps doesn't have an assert, so it just tries to remove it. However it is more work both here and in the testing to make sure they don't get added for non-alive machines, so just take the approach of "add one for all machines".

Reply:
You are also checking against life.Alive, which is a string, which won't match state.Life which is a uint8. This wasn't caught because the test was inserting what you were looking for rather than inserting a machine doc.